File: data/readable_format/cpp/file1658.cpp
#include <iostream>

namespace test { // Translating Fortran module to C++ namespace

    void oacc1() { // Translating Fortran subroutine to C++ function
        int i; // Translating Fortran integer to C++ int
        float pointee; // Translating Fortran real to C++ float
        float* ptr = &pointee; // Translating Fortran pointer to C++ pointer

        // Translating Fortran do loops to C++ for loops
        for(i = 1; i <= 5; i++) {
        }

        for(i = 1; i <= 5; i++) {
        }

        for(i = 1; i <= 5; i++) {
        }

        for(i = 1; i <= 5; i++) {
        }
    }

} // end of namespace test

int main() {
    test::oacc1(); // Calling the translated subroutine
    return 0;
}

File: data/readable_format/fortran/file1658.f90
module test
  implicit none
contains
  subroutine oacc1(result)
    integer, intent(out) :: result
    integer :: i
    result = 0
    do i = 1, 5
      result = result + i
    end do
  end subroutine oacc1
end module test

program unit_test
  use test
  implicit none
  integer :: result

  call oacc1(result)
  if (result == 15) then
    print *, "Test passed."
  else
    print *, "Test failed."
  end if
end program unit_test